src/components/flowchart/index.js

```diff
@@ -241,30 +265,62 @@ export class FlowChart extends Component {
   }
 
   /**
-   * Event handler for toggling a node's active state,
-   * showing tooltip, and highlighting linked nodes
+   * Enable a node's focus state and highlight linked nodes
+   * @param {Object} node Datum for a single node
+   */
+  handleNodeClick(node) {
+    this.props.onToggleNodeFocused(node.id);
+    this.setState({ nodeFocusActive: true });
+    event.stopPropagation();
+  }
+
+  /**
+   * Remove a node's focus state and dim linked nodes
+   */
+  handleChartClick() {
+    this.props.onToggleNodeFocused(null);
+    this.setState({ nodeFocusActive: false });
+  }
+
+  /**
+   * Enable a node's active state, show tooltip, and highlight linked nodes
    * @param {Object} node Datum for a single node
    */
   handleNodeMouseOver(node) {
-    const { layout, onToggleNodeActive } = this.props;
+    const { onToggleNodeActive, onToggleNodeFocused } = this.props;
     onToggleNodeActive(node, true);
     this.showTooltip(node);
-    linkedNodes.show({
-      el: this.el,
-      nodeID: node.id,
-      ...layout
-    });
+    if (!this.state.nodeFocusActive) {
+      onToggleNodeFocused(node.id);
+    }
   }
 
   /**
-   * Event handler for toggling a node's active state,
-   * hiding tooltip, and dimming linked nodes
+   * Remove a node's active state, hide tooltip, and dim linked nodes
    * @param {Object} node Datum for a single node
    */
   handleNodeMouseOut(node) {
     this.props.onToggleNodeActive(node, false);
-    linkedNodes.hide(this.el);
     this.hideTooltip();
+    if (!this.state.nodeFocusActive) {
+      this.props.onToggleNodeFocused(null);
+    }
+  }
+
+  /**
+   * Handle keydown event when a node is focused
+   * @param {Object} node Datum for a single node
+   */
+  handleNodeKeyDown(node) {
+    const ENTER = 13;
+    const ESCAPE = 27;
```

Comment:
Do you use key codes elsewhere in the app? Could be worth creating a helper map

Reply:
There's actually a helper in Kedro UI, and I considered using it, but felt that because this is such a short and simple bit of code, this is better because it's more explicit and less abstact, and replacing it with a helper would only make it more difficult to understand.

Reply:
Cool cool, suppose if we use key events again somewhere we can look at using the helper or making this more general purpose 👍